src/io/swiftsim_io.cpp

```diff
@@ -663,6 +663,9 @@ void SwiftSimReader_t::LoadSnapshot(MpiWorker_t &world, int snapshotId, vector<P
 
   Cosmology.Set(Header.ScaleFactor, Header.OmegaM0, Header.OmegaLambda0);
 
+  /* Assign the box size read in from the Header */
+  HBTConfig.BoxSize = Header.BoxSize;
```

Comment:
BoxSize is used to set BoxHalf before this point, so BoxHalf could be wrong if the box size in the header doesn't match the box size in the snapshot. If we're going to override parameter file entries we'll need to make sure they aren't used between the config file being parsed and us reading the snapshot.

Reply:
(BoxHalf is used in PeriodicDistance, which is defined in config_parser.h for some reason!)

Reply:
Good that you spotted that! I am a bit confused because of this, since that overwrite of the BoxSize seemed to fix haloes near the box edges in the tests I ran. Perhaps we should move the parameter overwrites to when the parameter file is initially being parsed, rather than doing it when the first snapshot is loaded. In the meantime, I will add a quick fix to this.

Reply:
I think PeriodicDistance will still work correctly even if BoxHalf is a bit wrong as long as all of the halos are much smaller than the box size.